src/inference/src/dev/core_impl.cpp

```diff
@@ -476,7 +476,7 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
                     desc.defaultConfig.erase(CONFIG_KEY(CACHE_DIR));
                 }
                 OPENVINO_SUPPRESS_DEPRECATED_END
-            }
+            });
```

Comment:
why do we need to guard here with `allowNotImplemented`?
If some metrics / properties are not inside `device_supports_cache_dir`, then we need to check it there

Reply:
This change was introduced by proxy plugin changes. Because initially mock plugins don't support some cache properties.
If some plugin doesn't support these properties we will get an exception here. 
Why should any plugin support these properties? If property doesn't exist it means taht it is not supported

Reply:
I mean that exception check should be moved closer to the calling code. Current peace of code does not look for any properties.